variables.tf

```diff
@@ -0,0 +1,218 @@
+variable "cloudwatch_logging_configuration" {
+  type = object({
+    alert_logs = optional(object({
+      enabled           = optional(bool, true)
+      log_group_prefix  = optional(string, "/platform/")
+      retention_in_days = optional(number, 90)
+    }), {})
+    flow_logs = optional(object({
+      enabled           = optional(bool, true)
+      log_group_prefix  = optional(string, "/platform/")
+      retention_in_days = optional(number, 90)
+    }), {})
+  })
+  default     = {}
+  description = "Cloudwatch logs configuration"
+}
+
+variable "delete_protection" {
+  type        = bool
+  description = "A boolean flag indicating whether it is possible to delete the firewall"
+  default     = true
+}
+
+variable "description" {
+  type        = string
+  description = "Description of the firewall instance"
+  default     = "AWS Network Firewall"
+}
+
+variable "fqdn_rules" {
+  type = map(object({
+    action    = string
+    fqdns     = list(string)
+    priority  = number
+    source_ip = list(string)
+  }))
+  default     = {}
+  description = "Map with L7 egress firewall rules"
+}
+
+variable "fqdn_rules_capacity" {
+  type        = number
+  description = "Capacity of the rule group"
+  default     = 1000
+}
+
+variable "fqdn_rule_group_priority" {
+  type        = number
+  description = "Priority of the rule group"
+  default     = 30
+}
+
+variable "home_net" {
+  type        = list(string)
+  description = "List of CIDRs to override the HOME_NET variable with"
+  default     = []
+}
+
+variable "ip_rules" {
+  type = map(object({
+    action           = string
+    destination_ip   = list(string)
+    destination_port = list(string)
+    direction        = string
+    priority         = number
+    protocol         = string
+    source_ip        = list(string)
+    source_port      = list(string)
+  }))
+  default     = {}
+  description = "Map with L4 firewall rules"
+}
+
+variable "ip_rules_capacity" {
+  type        = number
+  description = "Capacity of the rule group"
+  default     = 1000
+}
+
+variable "ip_rule_group_priority" {
+  type        = number
+  description = "Priority of the rule group"
+  default     = 40
+}
+
+variable "kms_key_arn" {
+  type        = string
+  default     = null
```

Comment:
You need to make this into a dynamic block:


  encryption_configuration {
    key_id = var.kms_key_arn
    type   = "CUSTOMER_KMS"
  }

or make this var required otherwise I think you will have apply errors. I think it makes sense to require a KMS key for now.